### PR TITLE
[COS-1665] Login with microsoft and adaption on current functions

### PR DESCRIPTION
**Pull request by grokkos**
## Changes

What types of changes does your code introduce?  _Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Build related changes
- [ ] Documentation Update (if none of the other choices apply)
- [ ] Other (please describe below)

## Summary by CodeRabbit

- **New Features**
	- Enhanced authentication to fetch user emails from Microsoft Graph API when not present in the profile.
	- Added support for Azure AD as an OAuth provider.

### CODE REVIEW

File: packages/apps/spaces/pages/api/auth/[...nextauth].ts [hunk 1]
```diff
@@ -38,6 +38,23 @@ export const authOptions: AuthOptions = {
         token.accessToken = account.access_token;
         token.id = (profile as GoogleProfile)?.id;
         token.playerIdentityId = account.providerAccountId;
+        token.email = (profile as GoogleProfile)?.email;
```
Comment (edifirut) [inline packages/apps/spaces/pages/api/auth/[...nextauth].ts]: Please remove line 41

**Review by edifirut — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/pages/api/auth/[...nextauth].ts [hunk 2]
```diff
@@ -38,6 +38,23 @@ export const authOptions: AuthOptions = {
...
+
+        // Check if the email is available in the profile
+        if (profile && profile.email) {
+          token.email = profile.email;
+        } else {
```
Comment (edifirut) [inline packages/apps/spaces/pages/api/auth/[...nextauth].ts]: Pleae check for azure-ad as provider. Otherwise we use it as fallback for google logins also

**Review by edifirut — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/pages/api/auth/[...nextauth].ts [hunk 3]
```diff
@@ -39,6 +39,22 @@ export const authOptions: AuthOptions = {
         token.id = (profile as GoogleProfile)?.id;
         token.playerIdentityId = account.providerAccountId;
 
+        // Check if the email is available in the profile
+        if (profile && profile.email && account.provider === 'azure-ad') {
```
Comment (edifirut) [inline packages/apps/spaces/pages/api/auth/[...nextauth].ts]: shouldn't be google here and azure on else?

**Review by edifirut — CHANGES_REQUESTED** [review]

File: packages/apps/spaces/pages/api/auth/[...nextauth].ts (hunk 3 above)
Comment (grokkos) [reply packages/apps/spaces/pages/api/auth/[...nextauth].ts]: can you check again?

**Review by edifirut — APPROVED** [review]

**Pull request merged**